core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -195,9 +208,30 @@ private synchronized void closeWithTragicEvent(Exception exception) throws IOExc
 
         operationCounter++;
 
+        assert assertSeqNoNotSeen(seqNo, data);
+
         return new Translog.Location(generation, offset, data.length());
     }
 
+    private boolean assertSeqNoNotSeen(long seqNo, BytesReference data) throws IOException {
```

Comment:
I know it's only called in one place but can you make it synchronized just for readability?